### ISSUE TITLE
Unable to sign transaction

### ISSUE THREAD

**Issue by congnghiakhiem** [mikemilano/xrphp#7]
```
>php test.php
PHP Fatal error:  Uncaught FOXRP\Rippled\Exception\RippledException: [0]: Unable to sign transaction: Unable to sign transaction: undefined:1
'{TransactionType:Payment,Account:rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz,Destination:raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2,Amount:1000000,Fee:000012,Sequence:1}'
^

SyntaxError: Unexpected token ' in JSON at position 0
    at JSON.parse (<anonymous>)
    at sign (xxx\npm\node_modules\xrpsign-cli\node_modules\xrpsign\index.js:8:19)
    at Object.<anonymous> (xxx\npm\node_modules\xrpsign-cli\index.js:12:28)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)


  thrown in xxx\vendor\foxrp\rippled-php\src\Api\Transaction.php on line 112
```

**Comment by mikemilano** [comment]
test.php?

**Comment by congnghiakhiem** [comment]
```
require __DIR__ . '/vendor/autoload.php';
use FOXRP\Rippled\Client;
use FOXRP\Rippled\Api\Transaction;

$client = new Client('https://s.altnet.rippletest.net:51234');

$txParams = [
    'TransactionType' => 'Payment',
    'Account' => 'rnJQ...oYUz',
    'Destination' => 'raEB...27d2',
    'Amount' => '1000000',
    'Fee' => '000012'
];

$transaction = new Transaction($txParams, $client);
$secret = 'snGK...HnG';
$response = $transaction->submit($secret);
```

**Comment by mikemilano** [comment]
Do you have `xrpsign-cli` installed?

See the `Local Signing Dependency` documentation here: https://github.com/foxrp/rippled-php/blob/master/docs/Transactions.md

**Comment by congnghiakhiem** [comment]
Of course, I have installed xrpsign-cli.
Please see: 
```
{TransactionType:Payment,Account:rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz,Destination:raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2,Amount:1000000,Fee:000012,Sequence:1}
```
I think the right JSON is:
```
{"TransactionType":"Payment","Account":"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz","Destination":"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2","Amount":1000000,"Fee":000012,"Sequence":1}
```

**Comment by mikemilano** [comment]
I think the quotes are being stripped for the sake of debug. I just checked and the command is run with the quotes. Also, the fee would require quotes because of the leading zeros.

Can you manually try running the xrpsign command from the command line? (replace SECRET with your secret)

`xrpsign '{"TransactionType":"Payment","Account":"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz","Destination":"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2","Amount":"1000000","Fee":"000012","Sequence":1}' 'SECRET'`

Just confirm it runs, and returns JSON in the command line. The response should look something like this:
> "tx":{"TransactionType":"Payment","Account":"rQBnNY5w5cALHbMaue2VefSzuBfxafwqp9","Destination":"rnQ1WgToG2RL9Fjmofif9ixYVgJTi6BLas","Amount":"1000000","Fee":"000012","Sequence":6,"SigningPubKey":"02F9637154C55935861AF4DF4FF1D0E9A21351D8436DD4C7792356DB2BA55B93E4","TxnSignature":"304402202DD395112350B16055CA68130B91FEBAAEF23FC5DC1F34B6A2F55FD1DB9AE21802205AB06DCDB7C0AAEB7E08C0102992D89653F0128AADC4C048C5A52A1B987F5D15"},"tx_blob":"12000024000000066140000000000F424068400000000000000C732102F9637154C55935861AF4DF4FF1D0E9A21351D8436DD4C7792356DB2BA55B93E47446304402202DD395112350B16055CA68130B91FEBAAEF23FC5DC1F34B6A2F55FD1DB9AE21802205AB06DCDB7C0AAEB7E08C0102992D89653F0128AADC4C048C5A52A1B987F5D158114FE32962E71441A81FB4FD80EE33E288A84FF5AB0831430643C3E4CCE37DD18F8AE238B7756A8CEC83FC5","tx_id":"E7FF719B850990F721F1513F6A6E492780CB497787F3220B58328756813C82C8"}

**Comment by mikemilano** [comment]
Alternatively, you could add some debug code to https://github.com/foxrp/rippled-php/blob/master/src/Api/Transaction.php#L90, by modifying the file in your vendor/foxrp directory.

That line builds the `xrpsign` command that `rippled-php` is trying to call. It's this try/catch block that is throwing the exception you're seeing.

Maybe print `$cmd` right after line 90, then try to run that command manually to see if it runs.

**Comment by mikemilano** [comment]
I think something is wrong with github tonight. Posts have gone missing.

**Comment by congnghiakhiem** [comment]
There is a problem with xrpsign
```
>xrpsign '{"TransactionType":"Payment","Account":"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz","Destination":"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2","Amount":"1000000","Fee":"000012","Sequence":1}' 'snGK...xHnG'
undefined:1
'{TransactionType:Payment,Account:rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz,Destination:raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2,Amount:1000000,Fee:000012,Sequence:1}'
^

SyntaxError: Unexpected token ' in JSON at position 0
    at JSON.parse (<anonymous>)
    at sign (...\npm\node_modules\xrpsign-cli\node_modules\xrpsign\index.js:8:19)
    at Object.<anonymous> (...\npm\node_modules\xrpsign-cli\index.js:12:28)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
```

**Comment by congnghiakhiem** [comment]
I think there is something wrong with xrpsign
```
...>xrpsign '{"TransactionType":"Payment","Account":"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz","Destination":"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2","Amount":"1000000","Fee":"000012","Sequence":1}' 'snGK...xHnG'
undefined:1
'{TransactionType:Payment,Account:rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz,Destination:raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2,Amount:1000000,Fee:000012,Sequence:1}'
^

SyntaxError: Unexpected token ' in JSON at position 0
    at JSON.parse (<anonymous>)
    at sign (...\AppData\Roaming\npm\node_modules\xrpsign-cli\node_modules\xrpsign\index.js:8:19)
    at Object.<anonymous> (...\AppData\Roaming\npm\node_modules\xrpsign-cli\index.js:12:28)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
```

**Comment by congnghiakhiem** [comment]
Hi Mike,

There is a problem with xrpsign:

C:\Users\Thalia>xrpsign
'{"TransactionType":"Payment","Account":"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz","Destination":"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2","Amount":"1000000","Fee":"000012","Sequence":1}'
'snGKpiGtbaswzXxkpvKwKmfrUxHnG'
undefined:1
'{TransactionType:Payment,Account:rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz,Destination:raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2,Amount:1000000,Fee:000012,Sequence:1}'
^

SyntaxError: Unexpected token ' in JSON at position 0
    at JSON.parse (<anonymous>)
    at sign
(C:\Users\Thalia\AppData\Roaming\npm\node_modules\xrpsign-cli\node_modules\xrpsign\index.js:8:19)
    at Object.<anonymous>
(C:\Users\Thalia\AppData\Roaming\npm\node_modules\xrpsign-cli\index.js:12:28)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)


Best Regards,
*Khiem Cong*
IT Researcher
Mobile: +84 90 22 11 560



Vào Th 2, 22 thg 10, 2018 vào lúc 14:52 Mike Milano <
notifications@github.com> đã viết:

**Comment by mikemilano** [comment]
I developed and tested it on a linux system. I've tested on OSX as well. I don't have a Windows system to test & debug it on.

I suspect that Windows does not like command line parameters to be wrapped in single quotes.

I'm going to have to look into this, unless you have any ideas.

**Comment by congnghiakhiem** [comment]
On Windows, need to passed json value like this:
```
xrpsign "{\"TransactionType\":\"Payment\",\"Account\":\"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz\",\"Destination\":\"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2\",\"Amount\":\"1000000\",\"Fee\":\"000012\",\"Sequence\":1}" "snGK...xHnG"
{"tx":{"TransactionType":"Payment","Account":"rnJQbpir8AJTdvidzGi31qpg9fqWJwoYUz","Destination":"raEBAeA8oc8pvKFPcEpQGWgp3TboU627d2","Amount":"1000000","Fee":"000012","Sequence":1,"SigningPubKey":"03929B6498D5F1AA963A931F7F137C63C0AE9EE8DAE0EFBAF87A8EC876AF162E6C","TxnSignature":"3044....362E"},"tx_blob":"1200...F8B1","tx_id":"7959CB51B77E6D09D3E46E614BAF7185A01EAA08D6FEF788BAD0D9FB2E6F02A1"}
```

**Comment by mikemilano** [comment]
@congnghiakhiem I've released version 1.0.1, which better uses the symfony/process abstraction.

It's unclear if it will resolve this, but please give it a try and report back your findings.

**Comment by congnghiakhiem** [comment]
It works ^^
